tests/test_console.py

```diff
@@ -39,3 +43,49 @@ def test_HBNBCommand_class_docstring(self):
                          "HBNBCommand class needs a docstring")
         self.assertTrue(len(HBNBCommand.__doc__) >= 1,
                         "HBNBCommand class needs a docstring")
+
+
+class TestHBNBCommandClass(TestCase):
+    """ Tests HBNBCommand class """
+
+    def test_do_create(self):
+        """ Tests create method """
+        with patch('sys.stdout', new=StringIO()) as f:
+            HBNBCommand().onecmd("count State")
+            state_count = int(f.getvalue())
+        with patch('sys.stdout', new=StringIO()) as f:
+            HBNBCommand().onecmd("count Place")
+            place_count = int(f.getvalue())
+        # checks error message if no class argument given
+        with patch('sys.stdout', new=StringIO()) as f:
+            HBNBCommand().onecmd("create")
+            self.assertEqual(f.getvalue(), "** class name missing **\n")
+        # checks error message if no class argument is invalid
+        with patch('sys.stdout', new=StringIO()) as f:
+            HBNBCommand().onecmd("create BadClassName")
+            self.assertEqual(f.getvalue(), "** class doesn't exist **\n")
+        # tests obj created successfully when first object
+        HBNBCommand().onecmd("create State name=\"California\"")
+        with patch('sys.stdout', new=StringIO()) as f:
+            HBNBCommand().onecmd("count State")
+            self.assertEqual(f.getvalue(), "{}\n".format(state_count + 1))
+        # tests obj created successfully when not first
+        HBNBCommand().onecmd("create State name=\"Nevada\"")
+        with patch('sys.stdout', new=StringIO()) as f:
+            HBNBCommand().onecmd("count State")
+            self.assertEqual(f.getvalue(), "{}\n".format(state_count + 2))
+        # tests obj created with different class, saves id
+        with patch('sys.stdout', new=StringIO()) as f:
+            HBNBCommand().onecmd(
+                "create Place name=\"My_little_house\" number_rooms=4")
+            p_id = f.getvalue()
+            p_id = p_id[:-1]
+        with patch('sys.stdout', new=StringIO()) as f:
+            HBNBCommand().onecmd("count Place")
+            self.assertEqual(f.getvalue(), "{}\n".format(place_count + 1))
+        # checks that after do_create, obj exists in dictionary
+        dict_key = "Place." + p_id
+        __objects = storage.all()
```

Comment:
I don't think it's necessary for this variable to be private.

Reply:
Yes, this doesn't necessarily have to be private, I was just trying to match the same variable references in the storage class (what the all method returns) to make it easier for anyone familiar with the code to switch back and forth.  To avoid any confusion, I'll update this to check for the key in storage.all().keys() more directly.

Reply:
Sounds good to me!